src/Functions/FunctionsConversion.h

```diff
@@ -303,7 +303,10 @@ struct ConvertImpl
                         }
                         else
                         {
-                            vec_to[i] = static_cast<ToFieldType>(vec_from[i]);
+                            if constexpr (std::is_same_v<ToDataType, DataTypeIPv4> && std::is_same_v<FromDataType, DataTypeUInt64>)
```

Comment:
so previously this conversion was valid and people just got incorrect data, if this uint64 was outside of 32-bit range?

Reply:
this static_cast makes narrowing conversion, so in clickhouse we are having implicit narrowing conversion - i.e. yes, if provided data out of range it will be narrowed loosing precision:
```
ip-172-31-42-195.us-east-2.compute.internal :) create table tbl (v UInt16) engine=Memory
Ok.

ip-172-31-42-195.us-east-2.compute.internal :) insert into tbl values (0x55555555)
Ok.

ip-172-31-42-195.us-east-2.compute.internal :) select * from tbl
┌─────v─┐
│ 21845 │
└───────┘
1 row in set. Elapsed: 0.005 sec.

ip-172-31-42-195.us-east-2.compute.internal :) select hex(v) from tbl
┌─hex(v)─┐
│ 5555   │
└────────┘
```
previously IPv4 type was an alias for UInt32, so this static_cast worked here, but now IPv4 is a dedicated type hidden inside StrongType wrapper, so it requires an additional static_cast for underlying type